cmd/dockerd/daemon.go

```diff
@@ -481,6 +482,12 @@ func initRouter(opts routerOptions) {
 		distributionrouter.NewRouter(opts.daemon.ImageService()),
 	}
 
+	for _, b := range []interface{}{opts.daemon, opts.buildBackend} {
+		if b, ok := b.(grpcrouter.Backend); ok {
+			routers = append(routers, grpcrouter.NewRouter(b))
```

Comment:
There would only be a single call to `grpcrouter.NewRouter`, just built up an array of backends and add it variadically. Adding the router multiple times is not right.